Size.pm

```diff
@@ -0,0 +1,68 @@
+package App::Ack::Filter::Size;
+
+use strict;
+use warnings;
+use base 'App::Ack::Filter';
+
+use Carp 'croak';
+
+use App::Ack::Filter ();
+
+sub _parse_size {
+    my $s = $_[0] || return 0;
+
+    if ( $s =~ m/^\s*(\d+(?:\.\d+)?)(?:\s*([KMGT]?)B?)?\s*$/i ) {
+        my $n = $1;
+        if ($2) {
+            my $u = lc $2;
+            $n *= 1024 while $u =~ tr/tgmk/gmk/d;
```

Comment:
This is pretty clever, but it might be *too* clever; maybe we should just have a hash of suffixes => number of bytes?